adapters/repos/db/vector/flat/index.go

```diff
@@ -708,3 +714,23 @@ func ValidateUserConfigUpdate(initial, updated schema.VectorIndexConfig) error {
 	}
 	return nil
 }
+
+func (h *flat) ShouldCompress() (bool, int) {
```

Comment:
flat index implements now new methods that seem not to be related to it.
are they going to be reimplemented in the future? if not maybe they should not be part of vector index interface.

Reply:
I agree, I have removed them and added an interface in the queue to detect indexes that support compression and ignore those who don't